dist/s_whitespace

```diff
@@ -10,9 +10,16 @@ trap 'rm -f $t' 0 1 2 3 13 15
 whitespace()
 {
     ! head $1 | grep -q 'automatically generated by SWIG' || return
-    (cat $1 && echo) | sed -e 's/[   ][      ]*$//' | \
-        cat -s | \
-        sed -e '${' -e '/^$/d' -e '}' > $t
+    if [[ "$1" == *.py ]]; then
+        # Special handling for Python, which allows multiple empty lines.
+        cat $1 | sed -e 's/[   ][      ]*$//' > $t
```

Comment:
This was originally a multi-platform way of saying `s/[[:space:]]+$//` for versions of `sed` that don't support features such as `+`. That being said, I have no idea why we wanted `+`; I will see if I can find a reason for it. If not, I agree it would make sense to convert it to `*`.

Good catch about the white-space in brackets. It used to be two characters, a tab followed by a space, until some commit inadvertently expanded those tabs into spaces, which I didn't catch. I'll check if `[[:space:]]` is supported on all our Evergreen platforms, or if I can do something else that specifies the tab as `\t`.

Reply:
I was able to use Evergreen to try `s_whitespace` with `[[:space:]]` on all platforms except RHEL 8 on zSeries and PPC, and it works, so I hope this means we can use it! (It works on RHEL 8 on x86, so I don't expect any surprises there.) So let's do that then.

And I didn't find any reason to use `+` instead of `*`, so I'm now using `s/[[:space:]]*$//` as you suggested.

Thank you!